cupy/__init__.py

```diff
@@ -119,29 +145,72 @@ def is_available():
 
 from numpy import uint64  # NOQA
 
+# -----------------------------------------------------------------------------
+# Floating-point numbers
+# -----------------------------------------------------------------------------
+from numpy import half  # NOQA
+
+from numpy import single  # NOQA
+
+from numpy import double  # NOQA
+
+from numpy import float_  # NOQA
+
+from numpy import longfloat  # NOQA
 
 from numpy import float16  # NOQA
 
 from numpy import float32  # NOQA
 
 from numpy import float64  # NOQA
 
+# Not supported by CuPy:
+# from numpy import float96
+# from numpy import float128
+
+# -----------------------------------------------------------------------------
+# Complex floating-point numbers
+# -----------------------------------------------------------------------------
+from numpy import csingle  # NOQA
 
 from numpy import complex_  # NOQA
 
+from numpy import clongfloat  # NOQA
+
 from numpy import complex64  # NOQA
 
 from numpy import complex128  # NOQA
 
-from numpy import csingle  # NOQA
+# Not supported by CuPy:
+# from numpy import complex192
+# from numpy import complex256
 
-from numpy import clongfloat  # NOQA
+# -----------------------------------------------------------------------------
+# Any Python object
+# -----------------------------------------------------------------------------
 
+# Not supported by CuPy:
+# from numpy import object_
+# from numpy import bytes_
+# from numpy import unicode_
+# from numpy import void
 
-from cupy.core import ufunc  # NOQA
+# -----------------------------------------------------------------------------
+# Built-in Python types
+# -----------------------------------------------------------------------------
 
-from numpy import newaxis  # == None  # NOQA
+from numpy import int  # NOQA
```

Comment:
Do you know is it documented in NumPy?
https://docs.scipy.org/doc/numpy-1.13.0/user/basics.types.html

Reply:
They're undocumented but seems intentionally exposed. https://github.com/cupy/cupy/issues/1043#issuecomment-374192273

Especially, `np.bool` is (mistakenly but actually) used instead of `np.bool_` in many codes.

Reply:
OK, I understood.